File: 2020/util/input.go
package util

import (
	"os"
	"strings"
)

func GetInput() string {
	input, err := os.ReadFile("input.txt")
	if err != nil {
		return ""
	}

	return strings.TrimSpace(string(input))
}

func GetInputLines() []string {
	return strings.Split(GetInput(), "\n")
}

File: 2018/src/util/input.js
const { readFileSync } = require('fs')

const getInput = () => {
  return readFileSync('input.txt', 'utf8').trim()
}

const getSplitInput = (delimiter = '\n') => {
  return getInput().split(delimiter)
}

const getInputLines = () => {
  return getSplitInput().map(l => l.trim())
}

module.exports = {
  getInput,
  getInputLines,
  getSplitInput,
}
